tests/integration/tables/groups.cpp

```diff
@@ -12,36 +12,52 @@
 
 #include <osquery/tests/integration/tables/helper.h>
 
+#include <osquery/utils/info/platform_type.h>
+
 namespace osquery {
 namespace table_tests {
+namespace {
 
 class groups : public testing::Test {
-  protected:
-    void SetUp() override {
-      setUpEnvironment();
-    }
+ protected:
+  void SetUp() override {
+    setUpEnvironment();
+  }
 };
 
 TEST_F(groups, test_sanity) {
-  // 1. Query data
-  auto const data = execute_query("select * from groups");
-  // 2. Check size before validation
-  // ASSERT_GE(data.size(), 0ul);
-  // ASSERT_EQ(data.size(), 1ul);
-  // ASSERT_EQ(data.size(), 0ul);
-  // 3. Build validation map
+  // Build validation map
   // See helper.h for avaialbe flags
   // Or use custom DataCheck object
-  // ValidatatioMap row_map = {
-  //      {"gid", IntType}
-  //      {"gid_signed", IntType}
-  //      {"groupname", NormalType}
-  //      {"group_sid", NormalType}
-  //      {"comment", NormalType}
-  //}
-  // 4. Perform validation
-  // validate_rows(data, row_map);
+  ValidatatioMap row_map = {
+      {"gid", IntType},
+      {"gid_signed", IntType},
+      {"groupname", NormalType},
+  };
+
+  if (isPlatform(PlatformType::TYPE_OSX)) {
+    row_map.emplace("is_hidden", IntType);
+  }
+
+  if (isPlatform(PlatformType::TYPE_WINDOWS)) {
+    row_map.emplace("comment", NormalType);
+    row_map.emplace("group_sid", NormalType);
+  }
+
+  // select * case
+  auto const rows = execute_query("select * from groups");
+  ASSERT_GE(rows.size(), 1ul);
+  validate_rows(rows, row_map);
+
+  // select with a specific gid
+  auto test_gid = rows.front().at("gid").c_str();
+  char query_string[50];
+  sprintf(query_string, "select * from groups where gid=%s", test_gid);
+  auto const rows_one = execute_query(query_string);
```

Comment:
I’m not sure if there is a reason you are constructing the string this way. 

You can concat two strings with +.

Reply:
Because I'm a c++ novice. Thanks for the catch!

Experimentally, I need to use `std::string("select * from groups where gid=") + test_gid`